src/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/DecryptTests.cs

```diff
@@ -21,7 +21,9 @@ namespace System.Security.Cryptography.Pkcs.EnvelopedCmsTests.Tests
 {
     public static partial class DecryptTests
     {
-        [Fact]
+        public static bool IsNet462OrNewer() => (!PlatformDetection.IsFullFramework || PlatformDetection.IsNetfx462OrNewer());
```

Comment:
> what I'm saying is that PlatformDetection.IsNetfx462OrNewer() should always return false on corefx 

It does, I believe.

> which implies that this logic should be moved to PlatformDetection.IsNetfx462OrNewer() directly

I don't follow. Given that IsNetfx462OrNewer returns `false` for corefx, but we want this to be `true` for corefx we need the "if I'm not desktop" clause.